[Docs] Update change log for v1.9.2 (#8306)
mlrun/mlrun: docs/change-log/index.md

- [Limitations](#limitations)
- [Deprecations and removed code](#deprecations-and-removed-code)

## Upgrading KFP, Python, and Pydantic
Upgrading these three MLRun dependencies spans several releases.  The upgrades are comprised of:

> - KFP: from 1.8 to 2.0. KFP has 2 components: the KFP service, and the KFP client code (which is used in both the MLRun service and various MLRun clients).

Reviewer: ```suggestion
- KFP: from 1.8 to 2.x. KFP has 2 components: the KFP service, and the KFP client package (which is used in both the MLRun service and some MLRun clients) and pipeline code (which is provided by the user).
```

- Pydantic: from version 1 to 2.
- Python: from 3.9 to 3.11.

This section describes the current guidelines. Specific changes are listed under the relevant versions.

### KFP
MLRun supports the KFP 2.x server, but workflows still require the KFP 1.8 syntax. Usage guidelines:
- Client code, workflow code and syntax (DSL) is still the KFP 1.8 syntax. Working with the newer KFP 2.x syntax is not yet supported by MLRun.

> - Starting with v1.8.0, KFP is not pre-installed on images such as `mlrun/mlrun`. The image `mlrun/mlrun-kfp` includes KFP, but works only with Python 3.9.

Reviewer: ```suggestion
- Starting with MLRun v1.8.0, KFP client package is not pre-installed on images such as `mlrun/mlrun`. The image `mlrun/mlrun-kfp` includes KFP, but works only with Python 3.9.
```
Just stating 1.8.0 may be confusing with the KFP version...

> - You can install KFP manually (`pip install kfp~=1.8`), for example, to run KFP pipelines locally using the KFP 1.8 client, which requires Python 3.9.

Reviewer: This instruction is different than what's written below (`pip install mlrun[kfp18]`). I actually think the other option is more correct since it adds MLRun packages that support kfp submission. Best to consult with @adamdelman or @alonmr on this.

### Python 
The MLRun server is based on a Python 3.11 environment. It's recommended to move the client to a Python 3.11 environment as well, although the MLRun client supports both Python 3.9 and Python 3.11.

- MLRun services (back-end) only use py3.11. 
- Client code and client-side images come out-of-the-box without KFP python packages installed. 

> MLRun provides two set of images: with Python 3.9 and with Python 3.11. You choose the image according to your dependencies and needs:

Reviewer: Maybe add an example here, explaining that the images will differ by tag, i.e. `mlrun/mlrun:1.9.2` vs. `mlrun/mlrun:1.9.2-py39`

  - The recommendation is to use the Python 3.11 images.

>   - MLRun attempts to deduce the correct Python version to use, based on the Python version where the user-code was written and submitted. So, if you work with Python 3.9 and submit a job to run in MLRun, MLRun uses the Python 3.9 image to ensure maximal compatibility.

Reviewer: ```suggestion
  - When running jobs, MLRun attempts to deduce the correct Python version to use, based on the Python version where the user-code was written and submitted. So, if you work with Python 3.9 and submit a job to run in MLRun which uses one of the built-in MLRun images, MLRun uses the Python 3.9 image to ensure maximal compatibility.
```

- MLRun provides an `mlrun-kfp` image that has KFP client pre-packaged in it. This image uses Python 3.9. The only intended usage for this image is for compiling user pipeline DSL code. See below for the usages of this image in the various MLRun execution modes. 
- If you are using Python 3.9 you have the option of compiling your workflow locally (meaning you are not working with a remote source). In this case make sure you installed mlrun with kfp (`pip install mlrun[kfp18]`).
- Workflows that are called by a client running 3.11 must use `engine="remote"`.

> When executing KFP pipelines, you choose whether to run the pipeline locally (with `kfp` engine) or remotely using a workflow runner (the `remote:kfp` engine). The following diagrams illustrate the recommended images for each use-case, and where Python 3.9 is needed. (The diagrams omit the argo pods that are launched by the KFP engine.)

Reviewer: Perhaps add a note here that the workflow runner is automatically created and managed by MLRun when using the `remote` engine.

- [Limitations](#limitations)
- [Deprecations and removed code](#deprecations-and-removed-code)

## Upgrading KFP, Python, and Pydantic
Upgrading these three MLRun dependencies spans several releases.  The upgrades are comprised of:

> - KFP: from 1.8 to 2.x. KFP has 2 components: the KFP service, and the KFP client package (which is used in both the MLRun service and some MLRun clients) and pipeline code (which is provided by the user).

Reviewer 2: the kfp client was not upgraded

| ID    |Description                                                                 |
|-------|----------------------------------------------------------------------------|
|ML-9326| MLRun now supports Python 3.11, and also continues to support Python 3.9. </br>Workflows that use Python 3.11 must use `engine="remote"`. |
|ML-10199|KFP 2.x server is now supported, but workflows still require the KFP 1.8 syntax. </br>Usage guidelines:<ul><li>Client code and workflow code and syntax (DSL) is still the KFP 1.8 syntax. Working with the newer KFP 2.x syntax is not yet supported by MLRun.</li><li>As in MLRun v1.8.0, KFP is not pre-installed on images such as `mlrun/mlrun`. The image `mlrun/mlrun-kfp` includes KFP, but works with Python 3.9.</li><li>You can install KFP manually (`pip install mlrun[kfp18]`), for example, to run KFP pipelines locally using the KFP 1.8 client, and thereby requiring Python 3.9.</li></ul>|

### Breaking change
| ID    |Description                                                                 |
|-------|----------------------------------------------------------------------------|

> |ML-10186|By default, the remote workflow runs with the mlrun/mlrun-kfp image that includes the KFP Python package. If you want to use a different image to compile your workflow, you must install kfp=~1.18 , use Python 3,9, and you can change the image by using `set_workflow(image=<image-name>)`.|

Reviewer 2: kfp~=1.8
